### PR TITLE
EdkRepo: Fix Checkout Pin

### LINKED ISSUE #142: Checkout pins is failing due to changed function parameters
checkout_repos() added one parameter: global_manifest_path

### LINKED ISSUE #145: Pin file names do not resolve properly
When using just the name of a pin file it is unable to be located in the global manifest repo. In this case .xml must be appended to the name to properly locate the file.

### PR DESCRIPTION
**Pull request by ashedesimone**
Add the global_manifest_path parameter to the
checkout_repos() call.

Corrects issue with pin resolution by
ensuring .xml is appended to args.pinfile if only
a name is used.

Additionally refactor the logic of __get_pin_path() so that it is easier to follow.

Fixes #142
Fixes #145

### CODE REVIEW

File: edkrepo/commands/checkout_pin_command.py [hunk 1]
```diff
@@ -98,23 +98,53 @@ def run_command(self, args, config):
...
+            pin_path = self.__find_pin_in_workspace(workspace_path, pin_name)
+
+        if pin_path:
+            return pin_path
         else:
             raise EdkrepoInvalidParametersException(humble.NOT_FOUND)
 
+
+    def __find_pin_in_manifest_repo(self, pin_name, manifest_repo_path, pin_path):
+        if not pin_name.endswith('.xml'):
```
Comment (kevinsun49) [inline edkrepo/commands/checkout_pin_command.py:121]: Is this check needed twice?

File: edkrepo/commands/checkout_pin_command.py [hunk 2]
```diff
@@ -98,23 +98,53 @@ def run_command(self, args, config):
...
-            return os.path.join(workspace_path, args.pinfile)
-        elif os.path.isfile(os.path.join(workspace_path, 'repo', args.pinfile)):
-            return os.path.join(workspace_path, 'repo', args.pinfile)
-        elif not os.path.isfile(os.path.join(workspace_path, args.pinfile)) and os.path.dirname(args.pinfile) is None:
-            for dirpath, dirnames, filenames in os.walk(workspace_path):
-                if args.pinfile in filenames:
-                    return os.path.join(dirpath, args.pinfile)
+            pin_path = os.path.normpath(args.pinfile)
+        elif manifest_repo_path is not None:
+            pin_path = self.__find_pin_in_manifest_repo(pin_name, manifest_repo_path, manifest.general_config.pin_path)
```
Comment (kevinsun49) [inline edkrepo/commands/checkout_pin_command.py:110]: We are losing the `os.path.normpath` for `manifest.general_config.pin_path`. Is this ok?

**Review by kevinsun49 — CHANGES_REQUESTED** [review]

File: edkrepo/commands/checkout_pin_command.py (hunk 2 above)
Comment (ashedesimone) [reply edkrepo/commands/checkout_pin_command.py:110]: Yes the __find_pin_in_manifest_repo() method will normalize the path before returning it.

File: edkrepo/commands/checkout_pin_command.py (hunk 1 above)
Comment (ashedesimone) [reply edkrepo/commands/checkout_pin_command.py:121]: No this input is sanitized prior to passing it to this method. I will submit an amendment.

**Review by kevinsun49 — APPROVED** [review]

**Pull request merged**